processor/spanmetricsprocessor/processor.go

```diff
@@ -350,24 +405,37 @@ func buildDimensionKVs(serviceName string, span pdata.Span, optionalDims []Dimen
 	return dims
 }
 
-func concatDimensionValue(metricKeyBuilder *strings.Builder, value string, prefixSep bool) {
+func buildResourceAttrKVs(serviceName string, optionalResourceAttrs []Dimension, resourceAttrs pdata.AttributeMap) dimKV {
```

Comment:
I think `extractResourceAttrsByKeys` might explicitly tell us what the function is doing. Also, renaming a few parameters might make the code easier to understand?
I'm thinking something like:
```go
func extractResourceAttrsByKeys(serviceName string, keys []Dimension, resourceAttrs pdata.AttributeMap) dimKV
```

btw could find a better name for the type `dimKV`? it could be confusing as we have the term `dimension` and `attributes`. How about we just call it `kvPairs`?

Reply:
In regards to the function name and parameter name changes I tried to keep it in line with what the original author had so that it's less confusing (see the function directly above this one). Should I modify those functions to have the same naming scheme as the proposed change you commented?

Reply:
I'll let you make the call. IMO I think we should rename them.

Reply:
ok, I'll rename it. To be honest I think we'll end up renaming this function in the PR where we add instrumentation library in since we need to aggregate by that too (I think not aggregating by it is currently kind of a bug).